packages/format-library/src/text-color/test/index.native.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * External dependencies
+ */
+import {
+	getEditorHtml,
+	initializeEditor,
+	fireEvent,
+	waitFor,
+} from 'test/helpers';
+
+/**
+ * WordPress dependencies
+ */
+import { setDefaultBlockName, unregisterBlockType } from '@wordpress/blocks';
+import { registerBlock, coreBlocks } from '@wordpress/block-library';
+
+const COLOR_PINK = '#f78da7';
+const paragraph = coreBlocks[ 'core/paragraph' ];
+
+const TEXT_WITH_COLOR = `<!-- wp:paragraph -->
+<p>Hello <mark style="background-color:rgba(0,0,0,0);color:#cf2e2e" class="has-inline-color has-vivid-red-color">this is a test</mark></p>
+<!-- /wp:paragraph -->`;
+
+beforeAll( () => {
+	registerBlock( paragraph );
+	setDefaultBlockName( paragraph.name );
+} );
+
+afterAll( () => {
+	unregisterBlockType( paragraph.name );
+} );
+
+describe( 'Text color', () => {
+	it( 'shows the text color formatting button in the toolbar', async () => {
+		const { getByA11yLabel } = await initializeEditor();
+
+		// Wait for the editor placeholder
+		const paragraphPlaceholder = await waitFor( () =>
+			getByA11yLabel( 'Add paragraph block' )
+		);
+		expect( paragraphPlaceholder ).toBeDefined();
+		fireEvent.press( paragraphPlaceholder );
+
+		// Wait for the block to be created
+		const paragraphBlock = await waitFor( () =>
+			getByA11yLabel( /Paragraph Block\. Row 1/ )
+		);
+		expect( paragraphBlock ).toBeDefined();
+
+		// Look for the highlight text color button
+		const textColorButton = await waitFor( () =>
+			getByA11yLabel( 'Text color' )
+		);
+		expect( textColorButton ).toBeDefined();
+	} );
+
+	it( 'allows toggling the highlight color feature to type new text', async () => {
+		const {
+			getByA11yLabel,
+			getByTestId,
+			getByA11yHint,
+		} = await initializeEditor();
+
+		// Wait for the editor placeholder
+		const paragraphPlaceholder = await waitFor( () =>
+			getByA11yLabel( 'Add paragraph block' )
+		);
+		expect( paragraphPlaceholder ).toBeDefined();
+		fireEvent.press( paragraphPlaceholder );
+
+		// Wait for the block to be created
+		const paragraphBlock = await waitFor( () =>
+			getByA11yLabel( /Paragraph Block\. Row 1/ )
+		);
+		expect( paragraphBlock ).toBeDefined();
+
+		// Look for the highlight text color button
+		const textColorButton = await waitFor( () =>
+			getByA11yLabel( 'Text color' )
+		);
+		expect( textColorButton ).toBeDefined();
+		fireEvent.press( textColorButton );
+
+		// Wait for Inline color modal to be visible
+		const inlineTextColorModal = getByTestId( 'inline-text-color' );
+		await waitFor( () => inlineTextColorModal.props.isVisible );
+
+		// Look for the pink color button
+		const pinkColorButton = await waitFor( () =>
+			getByA11yHint( COLOR_PINK )
+		);
+		expect( pinkColorButton ).toBeDefined();
+		fireEvent.press( pinkColorButton );
+
+		// Dismiss the inline color modal
+		fireEvent( inlineTextColorModal, 'backdropPress' );
```

Comment:
I noticed that the text color is applied when pressing a color so looks like we won't really need to dismiss the modal.